### PR TITLE
[aot] Reorganize graph argument creation process

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
Issue: #

### Brief Summary

Make it clean.

Note that it's intentionally not modifying the tests. Let's ensure the new interface doesn't lead to any breakage over the existing test cases.

### CODE REVIEW

File: python/taichi/examples/graph/texture_graph.py [hunk 1]
```diff
@@ -51,19 +51,14 @@ def main():
...
     g_init_builder.dispatch(make_texture, _rw_tex)
     g_init = g_init_builder.compile()
 
     g_init.run({'rw_tex': texture})
-    _tex = ti.graph.Arg(ti.graph.ArgKind.TEXTURE,
-                        'tex',
-                        channel_format=ti.f32,
-                        shape=(128, 128),
-                        num_channels=1)
+    _tex = ti.graph.Arg(ti.graph.ArgKind.TEXTURE, 'tex', ndim=2)
```
Comment (ailzhang) [inline python/taichi/examples/graph/texture_graph.py:61]: OOC: do we still need fmt info here?

File: python/taichi/graph/_graph.py [hunk 2]
```diff
@@ -97,77 +99,227 @@ def run(self, args):
...
+            if tag == ArgKind.RWTEXTURE:
+                fmt = TY_CH2FORMAT[(kwargs["channel_format"],
+                                    kwargs["num_channels"])]
+                kwargs["fmt"] = fmt
+                warnings.warn(
+                    "The channel_format and num_channels arguments for texture will be deprecated in v1.5.0, use fmt instead.",
+                    DeprecationWarning)
+            else:
+                warnings.warn(
+                    "The channel_format and num_channels arguments are only required for RW textures since v1.5.0, you can remove them safely.",
```
Comment (ailzhang) [inline python/taichi/graph/_graph.py:170]: are they "only required" or " no longer required" :P ?

File: python/taichi/graph/_graph.py [hunk 3]
```diff
@@ -97,77 +99,227 @@ def run(self, args):
...
+    if tag == ArgKind.RWTEXTURE or tag == ArgKind.TEXTURE:
+        if "dtype" in kwargs:
+            warnings.warn(
+                "The dtype argument for texture will be deprecated in v1.5.0, use format instead.",
+                DeprecationWarning)
+            del kwargs["dtype"]
+
+        if "shape" in kwargs:
+            warnings.warn(
+                "The shape argument for texture will be deprecated in v1.5.0, use ndim instead. (Note that you no longer need the exact texture size.)",
```
Comment (ailzhang) [inline python/taichi/graph/_graph.py:151]: OOC: can various textures with the same shape share the same taichi kernel?

**Review by ailzhang — APPROVED** [review]
Thanks!

File: python/taichi/graph/_graph.py (hunk 3 above)
Comment (PENGUINLIONG) [reply python/taichi/graph/_graph.py:151]: yes, we have struct for over rw textures

File: python/taichi/examples/graph/texture_graph.py (hunk 1 above)
Comment (PENGUINLIONG) [reply python/taichi/examples/graph/texture_graph.py:61]: only required for *rw* texture, not for textures

File: python/taichi/graph/_graph.py (hunk 2 above)
Comment (PENGUINLIONG) [reply python/taichi/graph/_graph.py:170]: nice catch

**Pull request merged**